cli/azd/pkg/tools/azcli/deployments_operations.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcli
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+)
+
+func (cli *azCli) createDeploymentsOperationsClient(
+	ctx context.Context,
+	subscriptionId string,
+) (*armresources.DeploymentOperationsClient, error) {
+	options := cli.createDefaultClientOptionsBuilder(ctx).BuildArmClientOptions()
+	client, err := armresources.NewDeploymentOperationsClient(subscriptionId, cli.credential, options)
+	if err != nil {
+		return nil, fmt.Errorf("creating deployments client: %w", err)
+	}
+
+	return client, nil
+}
+
+func (cli *azCli) ListSubscriptionDeploymentOperations(
+	ctx context.Context,
+	subscriptionId string,
+	deploymentName string,
+) (result []*armresources.DeploymentOperation, err error) {
+	result = make([]*armresources.DeploymentOperation, 0)
```

Comment:
If initializing with 0 entries we might as well just use `[]*armresources.DeploymentOperation{}`
```suggestion
	result = []*armresources.DeploymentOperation{}
```